src/main/xsl/preprocess/maprefImpl.xsl

```diff
@@ -389,6 +394,25 @@ See the accompanying LICENSE file for applicable license.
     </xsl:call-template>
   </xsl:template>
 
+  <xsl:template match="@*" mode="preserve-submap-attributes">
+    <xsl:attribute name="dita-ot:submap-{local-name()}" select="."/>
+  </xsl:template>
+  
+  <xsl:template match="*[contains(@class,' topic/title ')]" mode="preserve-submap-title-and-topicmeta">
+    <submap-topicmeta class="+ map/topicmeta ditaot-d/submap-topicmeta ">
+      <submap-title class="+ topic/navtitle ditaot-d/submap-title ">
+        <xsl:apply-templates select="@*" mode="preserve-submap-attributes"/>
+        <xsl:apply-templates select="*|processing-instruction()|text()"/>
+      </submap-title>
+    </submap-topicmeta>
+  </xsl:template>
+  <xsl:template match="*[contains(@class,' map/topicmeta ')]" mode="preserve-submap-title-and-topicmeta">
```

Comment:
Yes, moving inside is much cleaner. I wanted to keep the container because doing otherwise would change some results -- putting previously ignored `topicmeta` elements into the inheritance hierarchy, so that it would suddenly appear. The spec is generally quiet about whether this is right or wrong, so my goal was to make the elements _available_ but not push them in at this point. (This also removes any conflict concern about maps that have both `title` and `topicmeta/navtitle`.)